src/libsystemd-network/sd-radv.c

```diff
@@ -466,14 +466,14 @@ _public_ int sd_radv_set_hop_limit(sd_radv *ra, uint8_t hop_limit) {
         return 0;
 }
 
-_public_ int sd_radv_set_router_lifetime(sd_radv *ra, uint32_t router_lifetime) {
+_public_ int sd_radv_set_router_lifetime(sd_radv *ra, uint16_t router_lifetime) {
```

Comment:
I don't think we should do this. UINT_MAX is only ~18h, and we might want to set it larger than than in the future, even if the current underlying implementation does not allow this. So let's instead keep the original interface, and maybe return -ERANGE if appropriate.

Reply:
But RFC says the router lifetime is 16bit unsigned. We cannot easily extend it in the future. See e.g. [RFC 4191](https://tools.ietf.org/html/rfc4191#section-2.2), or netinet/icmp6.h.

Reply:
Seems shortsighted — I think it's fine for a field in a packet to be short (since bytes matter there). But using a 16 param in a function call, which is going to be passed as `int` anyway seems like a pointless optimization. But OK, I don't think this matters. If the kernel uses 16 bits, if this is ever changed, it'll need to be changed across the board.